docs/pages/application-access/guides/dynamic-registration.mdx

```diff
@@ -82,10 +82,27 @@ version: v5
 
 To create an application resource, run:
 
+<ScopedBlock scope={["oss", "enterprise"]}>
+
+```code
+# Log in to your Teleport cluster so you can use tctl remotely.
+# You can also run tctl on your Auth Service host.
+$ tsh login --proxy=teleport.example.com --user=myuser
+$ tctl create app.yaml
+```
+
+</ScopedBlock>
+<ScopedBlock scope={["cloud"]}>
+
 ```code
+# Log in to your Teleport cluster so you can use tctl remotely.
+$ tsh login --proxy=mytenant.teleport.sh --user=myuser
```

Comment:
This is just a meta thought but I wonder if there is a less verbose way for us to modify parts of the commands based on the selected scope. Right now we have to copy entire blocks when the only part that's different it the "proxy" value. E.g. can we have a "proxy" config variable that would change value based on the scope? Not sure if our engine supports that. Would also help with the consistency of examples as we'd use the same "example" proxy everywhere.

Reply:
Our variable replacement code is a [`remark`](https://github.com/remarkjs/remark) plugin that [replaces](https://github.com/gravitational/docs/blob/664b6a324c5d07e52ac1f4ecf76348e8d4c8cf84/server/remark-variables.ts#L47-L51) any instance of `(= varname =)` with the value provided in `config.json`. `remark` runs before the Markdown in our docs is converted into JSX (i.e., before scopes are assigned), so we would need to come up with a new way to populate variables based on scope.

I agree that we do have a lot of these `ScopedBlocks` that do nothing but rename the Proxy address, and it does get tedious to write and review. I'll create a ticket with the web team to look into this!